test/test_unit_keystore_credentials_provider.py

```diff
@@ -30,16 +35,66 @@ def test_load_credentials_do_not_exist_raises_credential_retrieval_exception(sel
         credential_provider = KeystoreCredentialsProvider()
         credential_provider.update_api_key_entry('someusertoupdate', MockCredentials, 'newapikey')
-        calls = [call('https://somemachine', 'machine', 'https://somemachine'), call('https://somemachine', 'login', 'someusertoupdate'), call('https://somemachine', 'password', 'newapikey') ]
+        calls = [call('https://somemachine', 'machine', 'https://somemachine'),
+                 call('https://somemachine', 'login', 'someusertoupdate'),
+                 call('https://somemachine', 'password', 'newapikey')]
         mock_store_adapter.assert_has_calls(calls)
 
     @patch.object(KeystoreAdapter, 'set_password', side_effect=Exception)
     def test_update_api_key_can_raise_operation_not_complete_exception(self, mock_store_adapter):
         with self.assertRaises(OperationNotCompletedException):
             credential_provider = KeystoreCredentialsProvider()
             credential_provider.update_api_key_entry('someusertoupdate', MockCredentials, 'newapikey')
+
+    @patch.object(KeystoreAdapter, "get_password", return_value=None)
+    def test_is_exists_return_false_if_attr_not_exists(self, mock_store_adapter):
+        credential_provider = KeystoreCredentialsProvider()
+        self.assertEquals(False, credential_provider.is_exists('https://somemachine'))
+
+    @patch.object(KeystoreAdapter, "get_password", return_value="password")
+    def test_is_exists_return_true_when_attr_exists(self, mock_store_adapter):
+        credential_provider = KeystoreCredentialsProvider()
+        self.assertEquals(True, credential_provider.is_exists('https://somemachine'))
+
+    @patch.object(KeystoreAdapter, "get_password", side_effect=Exception)
+    def test_is_exists_raises_exception_when_exception_is_thrown_by_get_password(self, mock_store_adapter):
+        credential_provider = KeystoreCredentialsProvider()
+        with self.assertRaises(Exception):
+            credential_provider.is_exists('https://somemachine')
+
+    @patch.object(KeystoreAdapter, 'delete_password', return_value=None)
+    def test_remove_credentials_calls_delete_password(self, mock_keystore_adapter):
+        mock_keystore_adapter.get_keyring_name.return_value = "keyring"
+        credential_provider = KeystoreCredentialsProvider()
+        conjurrc_data = ConjurrcData(conjur_url="https://conjur-hostname.com", account="admin")
+        credential_provider.remove_credentials(conjurrc_data)
+        calls = [
+            call("https://conjur-hostname.com"),
+        ]
+        mock_keystore_adapter.assert_has_calls(calls)
+
+    @patch.object(KeystoreAdapter, 'delete_password', side_effect=keyring.errors.KeyringError)
+    def test_remove_credentials_raises_keyring_error_when_delete_password_raises_keyring_error(self,
+                                                                                               mock_keystore_adapter):
+        mock_keystore_adapter.get_keyring_name.return_value = "keyring"
```

Comment:
think it will be cleaner if added as @patch.object at the top

Reply:
Done